src/nmr_to_structure/prepare_input/nmr_utils.py

```diff
@@ -1,44 +1,98 @@
+import datetime
 import os
+from functools import lru_cache
 from pathlib import Path
-from typing import Any, List, Optional, Tuple
+from typing import Any, Dict, KeysView, List, Optional, Tuple
 
 import numpy as np
 import pandas as pd
 import regex as re
-from rdkit import Chem
-from rdkit.Chem import rdMolDescriptors
+from rdkit import Chem, DataStructs
+from rdkit.Chem import AllChem, rdMolDescriptors
 from rxn.chemutils.tokenization import tokenize_smiles
+from rdkit import RDLogger
 from sklearn.model_selection import train_test_split
 
-RANDOM_SEED = 3246
+DEFAULT_SEED = 3246
+DEFAULT_NON_MATCHING_TOKEN = "<no_match> <no_match> <no_match> <no_match> <no_match> <no_match> <no_match> <no_match> <no_match> <no_match>"
+DEFAULT_ALLOWED_ELEMENTS = set(["C", "H", "N", "O", "S", "P", "F", "Cl", "Br", "I"])
```

Comment:
```suggestion
DEFAULT_ALLOWED_ELEMENTS = {"C", "H", "N", "O", "S", "P", "F", "Cl", "Br", "I"}
```
slightly more efficient